test/helpers/governance.new.js

```diff
@@ -0,0 +1,221 @@
+              proposal.id,
+              vote.support,
+              vote.voter,
+              vote.reason || '',
+              vote.params || '',
+              signature,
+            ),
+          )
+        : this.sign(vote).then(signature =>
+            this.governor.castVoteBySig(proposal.id, vote.support, vote.voter, signature),
+          )
+      : vote.params
+      ? // otherwise if params
+        this.governor.castVoteWithReasonAndParams(proposal.id, vote.support, vote.reason || '', vote.params)
+      : vote.reason
+      ? // otherwise if reason
+        this.governor.castVoteWithReason(proposal.id, vote.support, vote.reason)
+      : this.governor.castVote(proposal.id, vote.support);
+  }
+
+  sign(vote = {}) {
+    return vote.signature(this.governor, this.forgeMessage(vote));
+  }
+
+  forgeMessage(vote = {}) {
+    const proposal = this.currentProposal;
+
+    const message = { proposalId: proposal.id, support: vote.support, voter: vote.voter, nonce: vote.nonce };
+
+    if (vote.params || vote.reason) {
+      message.reason = vote.reason || '';
+      message.params = vote.params || '';
+    }
+
+    return message;
+  }
+
+  async waitForSnapshot(offset = 0n) {
+    const proposal = this.currentProposal;
+    const timepoint = await this.governor.proposalSnapshot(proposal.id);
+    return forward[this.mode](timepoint + offset);
+  }
+
+  async waitForDeadline(offset = 0n) {
+    const proposal = this.currentProposal;
+    const timepoint = await this.governor.proposalDeadline(proposal.id);
+    return forward[this.mode](timepoint + offset);
+  }
+
+  async waitForEta(offset = 0) {
+    const proposal = this.currentProposal;
+    const timestamp = await this.governor.proposalEta(proposal.id);
+    return forward.timestamp(timestamp.addn(offset));
+  }
+
+  /**
+   * Specify a proposal either as
+   * 1) an array of objects [{ target, value, data }]
+   * 2) an object of arrays { targets: [], values: [], data: [] }
+   */
+  setProposal(actions, description) {
+    let targets, values, data, useCompatibilityInterface;
+
+    if (Array.isArray(actions)) {
+      useCompatibilityInterface = actions.some(a => 'signature' in a);
+      targets = actions.map(a => a.target);
+      values = actions.map(a => a.value || '0');
+      data = actions.map(a => a.data || '0x');
+    } else {
+      ({ targets, values, data } = actions);
+    }
+
+    const descriptionHash = ethers.id(description);
+
+    // condensed version for queueing end executing
+    const shortProposal = [targets, values, data, descriptionHash];
+
+    // full version for proposing
+    const fullProposal = [targets, values, data, description];
+
+    // proposal id
+    const id = ethers.keccak256(
+      ethers.AbiCoder.defaultAbiCoder().encode(['address[]', 'uint256[]', 'bytes[]', 'bytes32'], shortProposal),
+    );
+
+    const currentProposal = {
+      id,
+      targets,
+      values,
+      signatures: [''],
```

Comment:
I realized we don't use this field anywhere even in the Truffle tests. I didn't remove it from the old `governance.js` file because we'll remove it but just mentioning for visibility